### PR TITLE
Split data binding page in docs

### PR DESCRIPTION
**Pull request by apedroferreira**
From https://github.com/mui/mui-toolpad/issues/1954

Split data binding section in "binding state" and "event handling" pages.

https://deploy-preview-1996--mui-toolpad-docs.netlify.app/toolpad/data-binding/binding-state/
https://deploy-preview-1996--mui-toolpad-docs.netlify.app/toolpad/data-binding/event-handling/

### CODE REVIEW

File: docs/data/toolpad/data-binding/binding-state.md
Comment (Janpot) [inline docs/data/toolpad/data-binding/binding-state.md:36]: Can you verify whether we're efficiently serving these videos? https://stackoverflow.com/a/10330501

Comment (prakhargupta1) [reply docs/data/toolpad/data-binding/binding-state.md:36]: Yes, in the HTTP headers following conditions are being met: Content-Type: video/mp4 and Accept-Ranges: bytes, and no Content-Encoding.

The video was recorded in dark theme and as discussed yesterday, it should be in light theme so, we'll have to remove this commit.

File: docs/data/toolpad/data-binding/binding-state.md
Comment (Janpot) [inline docs/data/toolpad/data-binding/binding-state.md:37]: Toolpad is in dark mode in this video

**Review by Janpot — APPROVED** [review]
Gonna approve already but left two comments with minor issues

**Comment by apedroferreira** [comment]
@prakhargupta1 Will you handle the rest of this PR? I can assign it to you.

Let me know if I can help with anything else here or if there's anything else I could add or fix here.

Also, we discussed that maybe "Data binding" should not be its own section, but be inside some other section?
In that case if we have a final decision I can move them inside the right section in this PR.
I guess we could put it under "Building UI" but I don't mind it being its own section either.

**Comment by Janpot** [comment]
In terms of docs structure, maybe we should just devote some sync meeting time and think about it holistically? In the meantime I don't think it should block merging this PR. It shouldn't be a big effort to rearrange pages afterwards.

**Comment by prakhargupta1** [comment]
@apedroferreira You can remove my commits as all of it was recorded in dark mode. I'll redo ot and create a separate PR.
Also, let's utilize coming Friday's product meeting to discuss docs structure.

File: docs/pages/toolpad/data-binding/binding-state.js
Comment (oliviertassinari) [inline docs/pages/toolpad/data-binding/binding-state.js:3]: Deleted pages need a redirection, fixed in #2083

**Pull request merged**
